extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java

```diff
@@ -1,16 +1,65 @@
 package io.quarkus.jackson.deployment;
 
+import java.util.Collection;
+
+import javax.inject.Inject;
+
+import org.jboss.jandex.*;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
 import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
 
 public class JacksonProcessor {
 
+    private static final DotName JSON_DESERIALIZE = DotName.createSimple(JsonDeserialize.class.getName());
+    private static final DotName BUILDER_VOID = DotName.createSimple(Void.class.getName());
+
+    private static final String JSON_DESERIALIZE_FIELD_BUILDER = "builder";
+
+    @Inject
+    BuildProducer<ReflectiveClassBuildItem> reflectiveClass;
+
+    @Inject
+    CombinedIndexBuildItem combinedIndexBuildItem;
+
     @BuildStep
-    void register(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
-        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false,
+    void register() {
+        addReflectiveClass(true, false,
                 "com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector",
-                "com.fasterxml.jackson.databind.ser.std.SqlDateSerializer"));
+                "com.fasterxml.jackson.databind.ser.std.SqlDateSerializer");
+
+        IndexView index = combinedIndexBuildItem.getIndex();
+
+        Collection<AnnotationInstance> pojoBuilderInstances = index.getAnnotations(JSON_DESERIALIZE);
+        for (AnnotationInstance pojoBuilderInstance : pojoBuilderInstances) {
+            addReflectiveClassInheritanceTree(pojoBuilderInstance.target().asClass());
```

Comment:
I am pretty sure this will fail badly when `@JsonDeserialize` is used to annotate anything but a class. Ideally we should handle all types, but maybe as a first step you can just check if the `AnnotationTarget.Kind` is a class

Reply:
Sorry to trouble you, but since only the `class` case is covered can you add a `TODO` note here explaining that we should support `@JsonDeserialize` on `methods`, `fields` etc.?
That way someone eager to  contribute when looking through the code can pick this up potentially